### DISCUSSION TITLE
Realtime user updates of a paticular room using pocketbase

### DISCUSSION THREAD

**Discussion by UtkarshSiddhpura** [pocketbase/pocketbase#3378] [Q&A] [answered]
I am creating a video chat app where users can create rooms and join or leave them. Is there a way for me to receive real-time updates when a user joins or leaves a specific room? In other words, can I subscribe to user changes for a particular room only?

I don't want the whole solution, but can you guide me in the right direction like how should i approach this?

**Comment by ganigeorgiev** [comment, marked as answer]
It depends what you understand by _"user changes for a particular room only"_ and how a _"user joins or leaves a specific room"_.

If you are looking for "presence" like behavior then this is not available out of the box (see also https://github.com/pocketbase/pocketbase/discussions/845) because it is difficult to generalize who will have access to that activity/presence information and how it will be managed.

There are not enough details to understand the full picture of your use case, but in general you can implement it on your own by using the available [Realtime API hooks](https://pocketbase.io/docs/go-event-hooks/#realtime-api-hooks) and broadcast to a custom realtime topic by calling something like `e.Client.Send(...)`.

Depending on your use case, an alternative and easier approach could be to create a new helper collection that you can subscribe to as usual and that will hold only the user and the room relations. In this case you can maintain the "presence" information by manually creating or deleting records in that collection.

_Eventually you can also combine both approaches, the hooks and the helper collection, to automate the user-room record create/delete operations when the client subscribe/disconnect._
